lib/dgram.js

```diff
@@ -283,20 +283,25 @@ function fixBufferList(list) {
 function enqueue(self, toEnqueue) {
   // If the send queue hasn't been initialized yet, do it, and install an
   // event handler that flushes the send queue after binding is done.
-  if (!self._sendQueue) {
-    self._sendQueue = [];
-    self.once('listening', function() {
-      // Flush the send queue.
-      for (var i = 0; i < this._sendQueue.length; i++)
-        this.send.apply(self, this._sendQueue[i]);
-      this._sendQueue = undefined;
-    });
+  if (!self._queue) {
+    self._queue = [];
```

Comment:
Yes, I wouldn't worry here, as it will get optimized on the long run, and
it does not add a huge delay during "boot" of a Socket.
Il giorno mar 31 mag 2016 alle 12:00 Ron Korving <notifications@github.com>
ha scritto:

> In lib/dgram.js
> <https://github.com/nodejs/node/pull/7066#discussion_r65153703>:
>
> > @@ -283,20 +283,25 @@ function fixBufferList(list) {
> >  function enqueue(self, toEnqueue) {
> >    // If the send queue hasn't been initialized yet, do it, and install an
> >    // event handler that flushes the send queue after binding is done.
> > -  if (!self._sendQueue) {
> > -    self._sendQueue = [];
> > -    self.once('listening', function() {
> > -      // Flush the send queue.
> > -      for (var i = 0; i < this._sendQueue.length; i++)
> > -        this.send.apply(self, this._sendQueue[i]);
> > -      this._sendQueue = undefined;
> > -    });
> > +  if (!self._queue) {
> > +    self._queue = [];
>
> The Socket class doesn't have a _queue property in its constructor. I
> didn't mean a deopt on this function, but on the hidden class for Socket.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nodejs/node/pull/7066/files/0e094acf5bed6a847c655fc69c33067e29227f60#r65153703>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AADL4wacLUEUMxSKPGvIT5WbfZubJhNZks5qHAaigaJpZM4Ip7BT>
> .
>